bin/handler-ec2_node.rb

```diff
@@ -138,23 +144,46 @@ def handle
     end
   end
 
+  # Method to delete client from sensu API
   def delete_sensu_client!
     response = api_request(:DELETE, '/clients/' + @event['client']['name']).code
     deletion_status(response)
   end
 
+  # Method to check if there is any insance and if instance is in a valid state that could be deleted
   def ec2_node_should_be_deleted?
+    # Defining region for aws SDK object
     ec2 = Aws::EC2::Client.new(region: region)
-    states = @event['client']['ec2_states'] || settings['ec2_node']['ec2_states'] || ['shutting-down', 'terminated', 'stopping', 'stopped']
+    # Check if attributes or json objects are not defined
+    if @event['client']['ec2_states'].nil? && settings['ec2_node']['ec2_states'].nil?
+      puts 'ec2_states is not define, please add the attributes or create a json config file with valid states keys'
```

Comment:
that is when ec2_states is not defined that was not there in the old script and I will just give an error so I put a print there so can see it in the sensu log.

This has been well tested and is backwards compatible BUT to make sure we are in the same page , the old script stopped working since the amazon api changed more than a year ago so that part is obviously updated.